### PR TITLE
Many things happened during the creation of this commit

### PR DESCRIPTION
**Pull request by pasandrei**
-implemented mAP API
-swapped the NMS function with a working one

### CODE REVIEW

File: data/dataloaders.py [hunk 1]
```diff
@@ -35,4 +35,4 @@ def get_dataloaders(params):
                                   shuffle=False, num_workers=4,
                                   sampler=BatchSampler(SequentialSampler([i for i in range(nr_images_in_val)]), batch_size=params.batch_size, drop_last=False))
 
-    return train_dataloader, valid_dataloader
```
Comment (RonaldGalea) [inline data/dataloaders.py]: this should be changed after testing

File: main.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ def run(path='misc/experiments/ssdnet/params.json', resume=False, eval_only=Fals
         start_epoch = checkpoint['epoch']
         print('Model loaded successfully')
 
-        for pg in optimizer.param_groups:
```
Comment (RonaldGalea) [inline main.py:53]: might as well delete this

File: misc/metrics.py [hunk 3]
```diff
@@ -136,7 +102,7 @@ def calculate_AP(model_output, label, anchors, grid_sizes, required_IoU=0.5):
     with torch.no_grad():
         for i in range(batch_size):
             prediction_bboxes, prediction_confidences = convert_output_to_workable_data(
-                model_output[0][i], model_output[1][i], anchors, grid_sizes)
+                model_output[0][i], model_output[1][i], anchors, grid_sizes, (320, 320))
```
Comment (RonaldGalea) [inline misc/metrics.py:105]: why (320, 320)?

File: train/validate.py [hunk 4]
```diff
@@ -37,22 +38,51 @@ def evaluate(model, optimizer, anchors, grid_sizes, train_loader, valid_loader,
...
+        for batch_idx, (input_, label, image_info) in enumerate(valid_loader):
             # print(datetime.datetime.now())
             input_ = input_.to(device)
             output = model(input_)
 
             batch_size = output[0].shape[0]
-            # for i in range(batch_size):
-            #     prediction_bboxes, predicted_confidences = convert_output_to_workable_data(
-            #         output[0][i], output[1][i], anchors, grid_sizes)
+            for i in range(batch_size):
```
Comment (RonaldGalea) [inline train/validate.py]: lines 49 - 78 should be a separate function in metrics and called from here

File: train/validate.py [hunk 5]
```diff
@@ -61,7 +91,20 @@ def evaluate(model, optimizer, anchors, grid_sizes, train_loader, valid_loader,
 
         SAVE_PATH = 'misc/experiments/{}/model_checkpoint'.format(params.model_id)
 
-        val_loss = (loc_loss_train + loc_loss_val) / val_set_size
+        with open("fisierul.json", 'w') as f:
```
Comment (RonaldGalea) [inline train/validate.py]: same here

File: main.py (hunk 2 above)
Comment (pasandrei) [reply main.py:53]: that line is deleted. What should I delete?

File: misc/metrics.py (hunk 3 above)
Comment (pasandrei) [reply misc/metrics.py:105]: Because that's how the model is trained ATM

**Review by RonaldGalea — APPROVED** [review]

**Pull request merged**